student/tests/views/test_registration_views.py

```diff
@@ -85,3 +88,45 @@ def test_registration_wizard_form_new_user(batch, existing_user):
     })
 
     assert User.objects.all().count() == 2
+
+def test_payment_preview_get_render_method_discount_logic():
```

Comment:
might want to be a bit more specific like `test_payment_preview_get_passes_discounted_price_to_render_if_referral_code_valid`